pkg/gatherers/clusterconfig/openshift_kube_apiserver_logs.go

```diff
@@ -0,0 +1,67 @@
+package clusterconfig
+
+import (
+	"context"
+
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openshift/insights-operator/pkg/gatherers/common"
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+// Extraction of dependency allows testing
+var kubeAPIServerCollectLogsFromContainers = common.CollectLogsFromContainers
+
+// GatherOpenShiftKubeAPIServerLogs collects logs from openshift-kube-apiserver:
+//   - if the log line is on error level
+//   - if the log line is on warning level and contains "error" substring
+//
+// The Kubernetes API:
+//
+//	https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/pod_expansion.go#L48
+//
+// Response see:
+//
+//	https://docs.openshift.com/container-platform/4.6/rest_api/workloads_apis/pod-core-v1.html#apiv1namespacesnamespacepodsnamelog
+//
+// * Location in archive: config/pod/{namespace-name}/logs/{pod-name}/errors.log
+// * Id in config: clusterconfig/openshift_kube_apiserver_logs
+// * Since version:
+//   - 4.12+
+func (g *Gatherer) GatherOpenShiftKubeAPIServerLogs(ctx context.Context) ([]record.Record, []error) {
+	gatherKubeClient, err := kubernetes.NewForConfig(g.gatherProtoKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+	coreClient := gatherKubeClient.CoreV1()
+
+	containersFilter := common.LogContainersFilter{
+		Namespace:                "openshift-kube-apiserver",
+		LabelSelector:            "app=openshift-kube-apiserver",
+		ContainerNameRegexFilter: "^kube-apiserver$",
+	}
+
+	messagesFilter := common.LogMessagesFilter{
+		MessagesToSearch: []string{
+			"E\\d+\\s.*",
+			"W\\d+\\s.*error.*",
```

Comment:
So I have following log snippet:
```
W1019 12:06:45.001360      16 watcher.go:229] watch chan error: etcdserver: mvcc: required revision has been compacted
{"level":"warn","ts":"2022-10-19T12:06:45.984Z","logger":"etcd-client","caller":"v3/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000999180/10.0.136.56:2379","attempt":0,"error":"rpc error: code = OutOfRange desc = etcdserver: mvcc: required revision has been compacted"}
```
The current regex patterns match only the first line. I think it would be good to have also the second line. WDYT pls?

Reply:
Current operator's logic is matching the patterns line by line ([here](https://github.com/openshift/insights-operator/blob/master/pkg/gatherers/common/gather_logs.go#L163))

This multi-line snippet probably was caused because the original log added a escaped \n (like \\n) on the log line to include more information.